clouddriver/clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/docker/HelmOciArtifactConfiguration.java

```diff
@@ -0,0 +1,75 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.clouddriver.artifacts.docker;
+
+import com.netflix.spinnaker.credentials.CredentialsTypeProperties;
+import com.netflix.spinnaker.kork.client.ServiceClientProvider;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import okhttp3.OkHttpClient;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@ConditionalOnExpression(
+    "${kubernetes.enabled:false} || ${cloudrun.enabled:false} || ${dockerRegistry.enabled:false}")
+@EnableConfigurationProperties({
+  HelmOciDockerArtifactProviderProperties.class,
+  HelmOciArtifactProviderProperties.class
+})
+@RequiredArgsConstructor
+@Slf4j
+class HelmOciArtifactConfiguration {
+
+  private final HelmOciDockerArtifactProviderProperties helmOciDockerArtifactProviderProperties;
+
+  private final HelmOciArtifactProviderProperties helmOciArtifactProviderProperties;
+
+  @Bean
+  public CredentialsTypeProperties<HelmOciDockerArtifactCredentials, HelmOciDockerArtifactAccount>
+      helmOciDockerCredentialsProperties(
+          OkHttpClient okHttpClient, HelmOciFileSystem helmChartsFileSystem) {
+    return CredentialsTypeProperties
+        .<HelmOciDockerArtifactCredentials, HelmOciDockerArtifactAccount>builder()
+        .type(HelmOciDockerArtifactCredentials.CREDENTIALS_TYPE)
+        .credentialsClass(HelmOciDockerArtifactCredentials.class)
+        .credentialsDefinitionClass(HelmOciDockerArtifactAccount.class)
+        .defaultCredentialsSource(helmOciDockerArtifactProviderProperties::getAccounts)
+        .credentialsParser(
+            a -> {
+              try {
+                return new HelmOciDockerArtifactCredentials(
+                    a, okHttpClient, helmChartsFileSystem, serviceClientProvider);
+              } catch (Exception e) {
+                log.warn("Failure instantiating Docker artifact account {}: ", a, e);
+                return null;
+              }
+            })
+        .build();
+  }
+
+  @Bean
+  public HelmOciFileSystem helmChartsFileSystem() {
+    return new HelmOciFileSystem(helmOciArtifactProviderProperties);
+  }
+
+  @Autowired ServiceClientProvider serviceClientProvider;
```

Comment:
Why not add a ServiceClientProvider argument to `helmOciDockerCredentialsProperties` ?  Maybe it doesn't matter, but it seems cleaner to me.

Reply:
Agreed, modified